com.eclipsesource.modelserver.emf/src/test/java/com/eclipsesource/modelserver/emf/ResourceManagerTest.java

```diff
@@ -0,0 +1,84 @@
+/********************************************************************************
+ * Copyright (c) 2019 EclipseSource and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+package com.eclipsesource.modelserver.emf;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+import java.io.IOException;
+import java.util.Optional;
+
+import org.eclipse.emf.ecore.EClass;
+import org.eclipse.emf.ecore.EObject;
+import org.eclipse.emf.ecore.EPackage;
+import org.eclipse.emf.ecore.resource.Resource;
+import org.eclipse.emf.ecore.util.EcoreUtil;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import com.eclipsesource.modelserver.emf.di.EMFModelServerModule;
+import com.google.common.collect.Lists;
+import com.google.inject.Guice;
+import com.google.inject.Injector;
+
+public class ResourceManagerTest extends AbstractResourceTest {
+
+	private static ResourceManager resourceManager;
+
+	@BeforeClass
+	public static void setupBeforeClass() {
+		Injector injector = Guice.createInjector(new EMFModelServerModule());
+		resourceManager = injector.getInstance(ResourceManager.class);
+	}
+
+	@Test
+	public void testLoadModel_castToExactType() {
+		String resourceURI = toFullPath("Test1.ecore");
+		Optional<EPackage> result = resourceManager.loadModel(resourceURI, resourceSet, EPackage.class);
+		resourceManager.saveAs(Lists.newArrayList(result.get()), toFullPath("Test1.json"));
+		assertNotNull(result);
+		assertTrue(result.isPresent());
```

Comment:
This assert is always true at this point (previous `get()`)

Reply:
Good catch. the `resourceManager.saveAs` method should be removed here. (Apparently, I was a bit sloppy on cleaning up the test cases :sweat_smile: )